### PR TITLE
Add image gallery to Core Content Page

### PR DESCRIPTION
**Pull request by hinzed1127**
Jira ticket: [LDAF-392](https://ldaf.atlassian.net/browse/LDAF-392)

## Proposed changes

- Adds two fields to the [Core content content model](https://app.contentful.com/spaces/pc5e1rlgfrov/content_types/serviceGroup/fields):
     1. Image Gallery Title (short text)
     2. Image Gallery (1-8 images, all of which must link to existing contentful assets)
 -  Queries for and renders image galleries on the core content page

## Screenshots

See the[ Indian Creek test page](https://ldaf-git-dan-image-gallery-ldaf.vercel.app/land/indian-creek)

## Acceptance criteria validation

- [x]  Matches MVP mocks
- [x]  Contentful model exists

### Alternate solutions

I opted to keep the image gallery markup within the core content `page.svelte`. It could be abstracted into a component if desired, but I'm not sure if we intent for this to be used in other contexts

## Requested feedback

<!-- If you plan on leaving a self-review, note that here. >

[LDAF-392]: https://ldaf.atlassian.net/browse/LDAF-392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.server.ts [hunk 1]
```diff
@@ -421,6 +452,10 @@ export const load = async ({
             }
           : undefined,
       },
+      imageGallery: {
+        images: imageGallery.filter(isNotNull),
```
Comment (hinzed1127) [inline src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.server.ts:456]: switch this to just `images: imageGallery` to see the TypeScript error I was encountering fixed by adding the filter method. If there's a cleaner way to do this in the query/type definitions above, I'd rather do that.

File: src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.server.ts [hunk 2]
```diff
@@ -305,7 +330,7 @@ export const load = async ({
     if (!pageMetadata) break fetchData;
     const client = getContentfulClient({
       spaceID: CONTENTFUL_SPACE_ID,
-      token: CONTENTFUL_DELIVERY_API_TOKEN,
+      token: CONTENTFUL_PREVIEW_API_TOKEN,
```
Comment (hinzed1127) [inline src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.server.ts:333]: TODO: switch this back to `CONTENTFUL_DELIVERY_API_TOKEN` before merging

File: src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.server.ts [hunk 3]
```diff
@@ -24,7 +33,11 @@ const baseQuery = gql`
   ${entryPropsFragment}
 
   query ServiceGroup($metadataID: String!) {
-    serviceGroupCollection(where: { pageMetadata: { sys: { id: $metadataID } } }, limit: 1) {
+    serviceGroupCollection(
+      where: { pageMetadata: { sys: { id: $metadataID } } }
+      limit: 1
+      preview: true
```
Comment (hinzed1127) [inline src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.server.ts:39]: TODO: remove preview flag before merging

File: src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.svelte [hunk 4]
```diff
@@ -143,6 +147,40 @@
...
 
+{#if imageGallery?.images.length > 0}
+  {#if imageGalleryTitle}
+    <h2>{imageGalleryTitle}</h2>
+  {/if}
+
+  <div>
+    <!-- Display a maximum of 8 images (2 rows of 4) -->
+    <!-- TODO: Enhance with a "show more" button revealing lightbox gallery -->
+    {#each chunk( imageGallery.images.flatMap((i) => [i, i]), 4, ).slice(0, 2) as row}
```
Comment (hinzed1127) [inline src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.svelte:158]: duplicating these for the purposes of seeing a full image gallery grid. will switch to this before merging
```suggestion
    {#each chunk( imageGallery.images, 4, ).slice(0, 2) as row}
```

**Review by hinzed1127 — COMMENTED** [review]
Self-review

**Review by Benaiah — APPROVED** [review]

**Pull request merged**